internal/webapp/vis/src/internal/D3Renderer.js

```diff
@@ -2926,6 +2926,146 @@ LABKEY.vis.internal.D3Renderer = function(plot) {
+                var dataObj = {};
+                dataObj[geom.xAes.value] = category;
+                dataObj[geom.xSubAes.value] = d[geom.xAes.value];
+
+                var value = d[category] ? d[category] : 0;
+                dataObj['value'] = value;
+                dataObj['y'] = +value;
+
+                return dataObj;
+            });
+        }));
+        data = stackData.reduce(function(prev, current) {
+            // keep track of which data points are from the last set, for the showValues case below
+            for (var index in prev) {
+                prev[index].showValue = false;
+            }
+            for (var index in current) {
+                current[index].showValue = true;
+            }
+            return prev.concat(current);
+        }, []);
+
+        binWidth = (plot.grid.rightEdge - plot.grid.leftEdge) / numXSubCategories;
+        barWidth = binWidth / 2;
+        offsetWidth = binWidth / 4;
+
+        hoverFn = geom.hoverFn ? geom.hoverFn : function(d) {
+            return (d.subLabel !== undefined ? 'Category: ' + d.subLabel + '\n' : '')
+                    + (d.label !== undefined ? 'Subcategory: ' + d.label + '\n' : '')
+                    + 'Value: ' + geom.yAes.getValue(d);
+        };
+
+        xAcc = function(d) {
+            return geom.getXSub(d) - offsetWidth;
+        };
+
+        yAcc = function(d){
+            return geom.getY(d)
+        };
+
+        colorAcc = geom.fill;
+        if (geom.colorAes && geom.colorScale)
+        {
+            colorAcc = function(row) {
+                return geom.colorScale.scale(geom.colorAes.getValue(row) + geom.layerName);
+            };
+        }
+
+        yZero = {};
+        yZero[geom.yAes.value] = 0;
+
+        // group each bar segment with an a tag for hover
+        barWrappers = layer.selectAll('a.bar-individual').data(data);
+        barWrappers.exit().remove();
+        barWrappers.enter().append('a').attr('class', 'bar-individual');
+        barWrappers.append('title').text(hoverFn);
+
+        // add the bars
+        rects = barWrappers.selectAll('rect.bar-rect').data(function(d){ return [d] });
+        rects.exit().remove();
+        heightFn = function(d) {
+            if (yAcc(d) === null || yAcc(d) === undefined) {
+                return 0;
+            }
+            return Math.abs(yAcc(d) - yAcc(yZero));
+        };
+        rects.enter().append('rect').attr('class', 'bar-rect')
+                .attr('x', function(d) {
+                    return xAcc(d);
+                })
+                .attr('y', function(d) {
+                    var verticalOffset = yAcc({ value: d.y0 });
+                    return verticalOffset - heightFn(d);
+                })
+                .attr('width', barWidth).attr('height', heightFn)
+                .attr('stroke', geom.color).attr('stroke-width', geom.lineWidth)
+                .attr('fill', colorAcc).attr('fill-opacity', geom.opacity);

```

Comment:
I feel like we should be more consistent with how we place calls to `.attr` but I don't know if that's how the styling is elsewhere in this code since it's been a long while for me.
```suggestion
                .attr('width', barWidth)
                .attr('height', heightFn)
                .attr('stroke', geom.color)
                .attr('stroke-width', geom.lineWidth)
                .attr('fill', colorAcc)
                .attr('fill-opacity', geom.opacity);

```

Reply:
yeah, doesn't look like it is very consistent across this file but I will at least clean it up for this new function